core/src/main/java/org/apache/carbondata/core/util/TrashUtil.java

```diff
@@ -157,11 +157,12 @@ public static void deleteExpiredDataFromTrash(String tablePath) {
     // Deleting the timestamp based subdirectories in the trashfolder by the given timestamp.
     try {
       if (FileFactory.isFileExist(trashPath)) {
```

Comment:
```suggestion
      CarbonFile trashFile = FileFactory.getCarbonFile(trashPath);
      if (trashFile.exists()) {
        CarbonFile[] timestampFolderList = trashFile.listFiles();
```

Reply:
I think we should deprecate listfiles and fileExist and other API in file factory that takes path, it has to take carbon file instead of path. This way we can force user to think about reusing the carbonFile Object.
we can raise a JIRA and assign some new contributors.

Reply:
currently Vikram can handle for his PR by accepting suggested changes by Indhu. But many places we are not resuing the CarbonFile object. that can be handled by my above point.

Reply:
https://issues.apache.org/jira/browse/CARBONDATA-4086

Reply:
done

Reply:
@ajantha-bhat , i can raise a PR for that as well